AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSCoreDataClient.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+#import "MSBaseOptions.h"
+#import "MSDataStore.h"
+#import "MSDocumentStore.h"
+#import "MSTokensResponse.h"
+#import "MS_Reachability.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^MSCachedTokenCompletionHandler)(MSTokensResponse *_Nullable tokens, NSError *_Nullable error);
+
+/**
+ * Class responsible for the core logic between offline and remote calls.
+ */
+@interface MSCoreDataClient : NSObject
+
+/**
+ * Data store.
+ */
+@property(nonatomic, nonnull) id<MSDocumentStore> documentStore;
+
+/**
+ * Network state helper.
+ */
+@property(nonatomic) MS_Reachability *reachability;
+
+/**
+ * Initialize a `MSCoreDataClient` instance.
+ *
+ * @param documentStore The document store instance.
+ *
+ * @return A new `MSCoreDataClient` instance.
+ */
+- (instancetype)initWithDocumentStore:(id<MSDocumentStore>)documentStore;
+
+/**
+ * Perform a core operation for a given partition/document
+ * using a combination of the local store and/or CosmosDB remote calls.
+ *
+ * @param operation The operation (nil, CREATE, UPDATE, DELETE).
```

Comment:
What does it mean to perform a `nil` operation?

Reply:
Updating documentation. The operation here is following what we store for pending operation in the document store. So `nil` = read operation.